server/src/main/java/com/saralein/server/router/Router.java

```diff
@@ -1,75 +1,35 @@
 package com.saralein.server.router;
 
 import com.saralein.server.controller.Controller;
-import com.saralein.server.controller.ErrorController;
-import com.saralein.server.protocol.Methods;
 import com.saralein.server.request.Request;
 import com.saralein.server.response.*;
-import java.nio.file.FileSystems;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
 
-public class Router {
+public class Router implements Controller {
```

Comment:
I am not sure I am into this. Is the `Router` really a `Controller`?  I think the issue is that we are missing an abstraction somewhere. In order to get polymorphism and satisfy the compiler, we have converted things in to controllers (`AuthMiddleware` -> `AuthController`, and `Router` -> `Router < Controller`).

What do you think?